### PR TITLE
Feature/uppsf 3099 use ontology models

### PR DESCRIPTION
**Pull request by ivan-p-nikolov**
# Description

## What

Refactor aggregate-concept-transformer service to use ontology concept models instead of the old static concept models.

## Why

https://financialtimes.atlassian.net/browse/UPPSF-3099

## Anything, in particular, you'd like to highlight to reviewers

Mention here sections of code which you would like reviewers to pay extra attention to .E.g

_Would appreciate a second pair of eyes on the test_  
_I am not quite sure how this bit works_  
_Is there a better library for doing x_  

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [ ] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

File: concept/healthcheck.go [hunk 1]
```diff
@@ -5,9 +5,13 @@ import (
...
 )
 
+type healthChecker interface {
+	Healthchecks() []fthealth.Check
+}
+
 type HealthService struct {
 	config *config
-	svc    Service
+	svc    healthChecker
```
Comment (ivanruski) [inline concept/healthcheck.go:14]: **thought:** We don't use `svc` anywhere except in the constructor. Maybe we should remove it from the struct.

File: concept/healthcheck.go [hunk 2]
```diff
@@ -5,9 +5,13 @@ import (
 	"github.com/Financial-Times/service-status-go/gtg"
 )
 
+type healthChecker interface {
+	Healthchecks() []fthealth.Check
+}
+
```
Comment (ivanruski) [inline concept/healthcheck.go:11]: **thought:** If we directly pass the `Healthchecks` in the constructor, we won't even need this interface.

File: concept/service.go [hunk 3]
```diff
@@ -257,14 +253,19 @@ func (s *AggregateService) ProcessMessage(ctx context.Context, UUID string, book
...
 	if concordedConcept.Type == "Membership" {
-		if err = sendToPurger(ctx, s.httpClient, s.varnishPurgerAddress, []string{concordedConcept.PersonUUID}, "Person", s.typesToPurgeFromPublicEndpoints, transactionID); err != nil {
-			logger.WithTransactionID(transactionID).WithUUID(concordedConcept.PersonUUID).Errorf("Concept couldn't be purged from Varnish cache")
+		personUUID, err := getPersonUUIDFromConcept(concordedConcept)
+		if err != nil {
+			logger.WithTransactionID(transactionID).WithUUID(concordedConcept.PrefUUID).WithError(err).Errorf("Concept couldn't be purged from Varnish cache")
+		}
+		err = sendToPurger(ctx, s.httpClient, s.varnishPurgerAddress, []string{personUUID}, "Person", s.typesToPurgeFromPublicEndpoints, transactionID)
+		if err != nil {
+			logger.WithTransactionID(transactionID).WithUUID(personUUID).Errorf("Concept couldn't be purged from Varnish cache")
```
Comment (ivanruski) [inline concept/service.go:268]: **nitpick:** I think we've missed `.WithError`

**Review by ivanruski — DISMISSED** [review]

File: concept/service_test.go [hunk 4]
```diff
@@ -1232,6 +1218,26 @@ func TestResolveConceptType(t *testing.T) {
 	assert.Equal(t, "organisations", company)
 }
 
+// getConceptFromService is a wrapper function for getting concepts from the service
+//
+// nolint: revive, unparam
```
[the remark this answers is not in the document]
Comment (ivan-p-nikolov) [reply concept/service_test.go:1223]: As the warning in CircleCI states `golint` is deprecated and `revive` should be used

File: concept/service.go (hunk 3 above)
Comment (ivan-p-nikolov) [reply concept/service.go]: fixed

File: concept/healthcheck.go (hunk 1 above)
Comment (ivan-p-nikolov) [reply concept/healthcheck.go:14]: True. I didn't even look how `svc` was used. My aim was not to refactor the service too much.

I think I'll leave it as it is.

If we want to change `HealthService` it should be bigger refactor. For instance there is no reason `AggregateService` to have `Healthchecks` method. All that logic could and should be separate from `AggregateService`

**Review by ivanruski — DISMISSED** [review]

File: concept/service_test.go [hunk 5]
```diff
@@ -1232,6 +1218,26 @@ func TestResolveConceptType(t *testing.T) {
...
+func getConceptFromService(svc *AggregateService, ctx context.Context, conceptUUID string, bookmark string) (transform.OldAggregatedConcept, string, error) {
+	c, tid, err := svc.GetConcordedConcept(ctx, conceptUUID, bookmark)
+	if err != nil {
+		return transform.OldAggregatedConcept{}, "", err
+	}
+	old, err := transform.ToOldAggregateConcept(c)
+	if err != nil {
+		return transform.OldAggregatedConcept{}, "", err
+	}
+	sort.Strings(old.Aliases)
```
Comment (ManoelMilchev) [inline concept/service_test.go:1236]: `praise`: It's cool that you've moved the sorting here!

**Review by ManoelMilchev — APPROVED** [review]
LGTM!

**Review by ManoelMilchev — APPROVED** [review]
LGTM!

**Pull request merged**